Marlin/src/feature/leds/neopixel.cpp

```diff
@@ -36,10 +36,20 @@
 
 Adafruit_NeoPixel pixels(NEOPIXEL_PIXELS, NEOPIXEL_PIN, NEOPIXEL_TYPE + NEO_KHZ800);
 
+#ifdef NEOPIXEL_BKGD_LED_INDEX
+  void set_neopixel_color_background() {
+    uint8_t background_color[4] = NEOPIXEL_BKGD_COLOR;
+    pixels.setPixelColor(NEOPIXEL_BKGD_LED_INDEX, pixels.Color(background_color[0], background_color[1], background_color[2], background_color[3]));
+  }
+#endif
+
 void set_neopixel_color(const uint32_t color) {
   for (uint16_t i = 0; i < pixels.numPixels(); ++i) {
     #ifdef NEOPIXEL_BKGD_LED_INDEX
-      if (NEOPIXEL_BKGD_LED_INDEX == i) i++;
+      if (i == NEOPIXEL_BKGD_LED_INDEX && color != 0x000000) {
+        set_neopixel_color_background();
+        continue;
+      }
```

Comment:
This breaks the previous functionality, where the loop skips the background LED index. This would cause *any* call to `set_neopixel_color` to set the background color of the LCD, which is not what we want! (Note that the original code that does `i++` was incorrect and should have been `continue`.)

Reply:
Q: Is this logic needed for all cases of `NEOPIXEL_BKGD_LED_INDEX` or just for `LED_BACKLIGHT_TIMEOUT`?

Reply:
This seems like it breaks the previous functionality, where the loop skips the background LED index. As this is here, *any* call to `set_neopixel_color` will now update the background of the LCD.

Q: Is this logic needed for all cases of `NEOPIXEL_BKGD_LED_INDEX` or just for `LED_BACKLIGHT_TIMEOUT`?